kolibri/core/assets/src/state/actions.js

```diff
@@ -223,6 +223,36 @@ function getCurrentSession(store) {
   .catch(error => { handleApiError(store, error); });
 }
 
+function getFacilityConfig(store) {
+  const coreApp = require('kolibri');
+  const FacilityCollection = coreApp.resources.FacilityResource
+    .getCollection()
+    .fetch();
+
+  return FacilityCollection.then(facilities => {
+    const currentFacilityId = facilities[0].id;
+    const facilityConfigCollection = coreApp.resources.FacilityDatasetResource
+      .getCollection({ facility_id: currentFacilityId })
+      .fetch();
+    return facilityConfigCollection
```

Comment:
Yes, I think reducing the chain is good - I am also happy to just use snake_cased attributes directly on the client side in order to take away another step, and make it more transparent.

I also think that, as far as possible, the data that makes it to the client should be in the form the client wants it in (to this end, the JSON fields that we use in some places will be transitioned to be properly turned into JSON before leaving the server, rather than getting to the client side in a text form that then has to be JSON parsed).

I think filtering the API data through state mappers is not a great pattern, precisely for the reason you describe, and instead, we should more broadly implement support for the `fields` getParam that allows us to explicitly specify the fields that get returned (https://github.com/learningequality/kolibri/blob/develop/kolibri/content/serializers.py#L59).